tests/test_lib_file_analyse.py

```diff
@@ -0,0 +1,81 @@
+import os
+from wheel.lib_file_analyse import extract_macosx_min_system_version
+from wheel.pep425tags import get_platform
+
+
+def test_read_from_dynlib():
+    dirname = os.path.dirname(__file__)
+    dylib_dir = os.path.join(dirname, "testdata",
+                             "macos_minimal_system_version")
+    versions = [
+        ("test_lib_10_6_fat.dynlib", "10.6"),
+        ("test_lib_10_10_fat.dynlib", "10.10"),
+        ("test_lib_10_14_fat.dynlib", "10.14"),
+        ("test_lib_10_6.dynlib", "10.6"),
+        ("test_lib_10_10.dynlib", "10.10"),
+        ("test_lib_10_14.dynlib", "10.14"),
+        ("test_lib_10_6_386.dynlib", "10.6"),
+        ("test_lib_10_10_386.dynlib", "10.10"),
+        ("test_lib_10_14_386.dynlib", "10.14"),
+        ("test_lib_multiple_fat.dynlib", "10.14")
+    ]
+    for file_name, ver in versions:
+        extracted = extract_macosx_min_system_version(
+            os.path.join(dylib_dir, file_name)
+        )
+        str_ver = str(extracted[0]) + "." + str(extracted[1])
+        assert str_ver == ver
+        assert extract_macosx_min_system_version(
+            os.path.join(dylib_dir, "test_lib.c")
+        ) is None
+
+
+def test_get_platform_macos(mocker, capsys):
+    print(mocker, mocker.patch, mocker.patch.mock_module)
+    dirname = os.path.dirname(__file__)
+    dylib_dir = os.path.join(dirname, "testdata",
+                             "macos_minimal_system_version")
+    with mocker.patch("distutils.util.get_platform", return_value="macosx-10.14-x86_64"):
```

Comment:
If you're using mocks in `with` blocks, what's the point of using mocker in the first place?

Reply:
I'm learning how to use mock. If I good understand in python 2.7 there is preference to use mock library and  in python 3.4+ use `unittests.mock`. 

And I do not have idea how test this functions without mock. 

How to fix it in best way. change dependency on `mock`. Do some if depends on python version?

Reply:
Python 2.6 did not have `unittest.mock`. But Python 2.7 does. The `mocker` fixture, on the other hand, works like pytest's built-in `monkeypatch` fixture in that it automatically undoes all the mocks that it sets up during the test request.

Reply:
I rewrite tests to base on `monkeypatch`. First I think that it is not enough, but maybe I better understand it.